sources/s3/s3.go

```diff
@@ -0,0 +1,520 @@
+		ownershipControls, _ = client.GetBucketOwnershipControls(ctx, &s3.GetBucketOwnershipControlsInput{Bucket: bucketName})
+	}()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		policy, _ = client.GetBucketPolicy(ctx, &s3.GetBucketPolicyInput{Bucket: bucketName})
+	}()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		policyStatus, _ = client.GetBucketPolicyStatus(ctx, &s3.GetBucketPolicyStatusInput{Bucket: bucketName})
+	}()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		replication, _ = client.GetBucketReplication(ctx, &s3.GetBucketReplicationInput{Bucket: bucketName})
+	}()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		requestPayment, _ = client.GetBucketRequestPayment(ctx, &s3.GetBucketRequestPaymentInput{Bucket: bucketName})
+	}()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		tagging, _ = client.GetBucketTagging(ctx, &s3.GetBucketTaggingInput{Bucket: bucketName})
+	}()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		versioning, _ = client.GetBucketVersioning(ctx, &s3.GetBucketVersioningInput{Bucket: bucketName})
+	}()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		website, _ = client.GetBucketWebsite(ctx, &s3.GetBucketWebsiteInput{Bucket: bucketName})
+	}()
+
+	// Wait for all requests to complete
+	wg.Wait()
+
+	bucket := Bucket{
+		// TODO: Where do we get this from??
+		Bucket: types.Bucket{},
+
+		GetBucketAclOutput:                             *acl,
+		GetBucketAnalyticsConfigurationOutput:          *analyticsConfiguration,
+		GetBucketCorsOutput:                            *cors,
+		GetBucketEncryptionOutput:                      *encryption,
+		GetBucketIntelligentTieringConfigurationOutput: *intelligentTieringConfiguration,
+		GetBucketInventoryConfigurationOutput:          *inventoryConfiguration,
+		GetBucketLifecycleConfigurationOutput:          *lifecycleConfiguration,
+		GetBucketLocationOutput:                        *location,
+		GetBucketLoggingOutput:                         *logging,
+		GetBucketMetricsConfigurationOutput:            *metricsConfiguration,
+		GetBucketNotificationConfigurationOutput:       *notificationConfiguration,
+		GetBucketOwnershipControlsOutput:               *ownershipControls,
+		GetBucketPolicyOutput:                          *policy,
+		GetBucketPolicyStatusOutput:                    *policyStatus,
+		GetBucketReplicationOutput:                     *replication,
+		GetBucketRequestPaymentOutput:                  *requestPayment,
+		GetBucketTaggingOutput:                         *tagging,
+		GetBucketVersioningOutput:                      *versioning,
+		GetBucketWebsiteOutput:                         *website,
+	}
+
+	attributes, err := sources.ToAttributesCase(bucket)
+
+	if err != nil {
+		return nil, &sdp.ItemRequestError{
+			ErrorType:   sdp.ItemRequestError_OTHER,
+			ErrorString: err.Error(),
+			Scope:       scope,
+		}
+	}
+
+	item := sdp.Item{
+		Type:            "s3-bucket",
+		UniqueAttribute: "name",
+		Attributes:      attributes,
+	}
+
+	if bucket.RedirectAllRequestsTo != nil {
```

Comment:
how can `RedirectAllRequestsTo` be ever not nil?

Reply:
When I create the struct I'm using [composition](https://golangbot.com/inheritance/) to make the `Bucket` type basically like a mashed up union of all the `Get*Output` types:

```go
type Bucket struct {
	// ListBuckets
	types.Bucket

	s3.GetBucketAclOutput
```

Doing this promotes both the fields and methods of the composed struct to the parent one, like inheritance (sort of). So since `s3.GetBucketWebsiteOutput` has a Field called `RedirectAllRequestsTo`, my local `Bucket` type will have that field too. You set the fields in a kind of weird way as you can see on line 275, but you can pull them back as if they were native. Intellisense picks all this up to so it gets autocompleted nicely